### ISSUE TITLE
A more dynamic front page

### ISSUE THREAD

**Issue by tclune** [Fortran-FOSS-Programmers/ford#471]
One of the more interesting comments from our review was a wish to have a different view on the main page.   Rather than seeing say the first dozen of the global lists of source files, modules, etc. which are often just a tiny fraction of the project, the suggestion was instead that the main first column be "Modules" (unavoidably long).   But that the "Procedures" and "Derived Types" columns would be  dynamically generated as you select an item in the "Modules" column.  I.e., the "Procedures" and "Derived Types" would only display the names of the procedures and types defined in the selected module.     Not only do we believe that this would simplify navigation in very large projects, it also helps to avoid ambiguity when the same procedure name is used in multiple modules.  E.g.,  "get()" and other accessors tend to get reused a lot making it hard to select them from a global list.

Related to this is the desire to have the option to suppress the "Source Files" column entirely.   This aspect is presumably a relatively trivial change.

**Comment by ZedThree** [comment]
Controlling exactly which columns appear on the front page should definitely be trivial.

I'm not sure about the dynamic generation though -- we might need a proper web dev to implement something like that.

Maybe something easier would be to keep it static but copy how something like the sphinx-book-theme does it -- see the [FORD docs](https://forddocs.readthedocs.io/en/latest/api/ford.html) for instance

**Comment by tclune** [comment]
Yeah - this is well beyond any web skills that I envisioning for myself.  

Adding @mathomp4 and @JulesKouatchou  to monitor/comment this issue.
